.test-infra/jenkins/dependency_check/dependency_check_report_generator.py

```diff
@@ -0,0 +1,279 @@
+      elif compare_dependency_release_dates(curr_release_date, latest_release_date):
+        high_priority_deps.append(dep_info)
+    except:
+      traceback.print_exc()
+      continue
+
+  return high_priority_deps
+
+
+def compare_dependency_versions(curr_ver, latest_ver):
+  """
+  Compare the current using version and the latest version.
+  Return true if a major version change was found, or 3 minor versions that the current version is behind.
+  Args:
+    curr_ver
+    latest_ver
+  Return:
+    boolean
+  """
+  if curr_ver is None or latest_ver is None:
+    return True
+  else:
+    curr_ver_splitted = curr_ver.split('.')
+    latest_ver_splitted = latest_ver.split('.')
+    curr_major_ver = curr_ver_splitted[0]
+    latest_major_ver = latest_ver_splitted[0]
+    # compare major versions
+    if curr_major_ver != latest_major_ver:
+      return True
+    # compare minor versions
+    else:
+      curr_minor_ver = curr_ver_splitted[1] if len(curr_ver_splitted) > 1 else None
+      latest_minor_ver = latest_ver_splitted[1] if len(latest_ver_splitted) > 1 else None
+      if curr_minor_ver is not None and latest_minor_ver is not None:
+        if (not curr_minor_ver.isdigit() or not latest_minor_ver.isdigit()) and curr_minor_ver != latest_minor_ver:
+          return True
+        elif int(curr_minor_ver) + _MAX_MINOR_VERSION_DIFF <= int(latest_minor_ver):
+          return True
+  return False
+
+
+def query_dependency_release_dates(bigquery_client, dep_name, curr_ver, latest_ver):
+  """
+  Query release dates of current using version and the latest version from BQ tables.
+  Args:
+    bigquery_client: a bq client object that bundle configurations for API requests
+    dep_name: dependency name
+    curr_ver: the current using version
+    latest_ver: the later version
+  Return:
+    curr_release_date
+    latest_release_date
+  """
+  try:
+    curr_release_date = bigquery_client.query_dep_release_date(dep_name, curr_ver)
+    latest_release_date = bigquery_client.query_dep_release_date(dep_name,latest_ver)
+    if curr_release_date is None:
+      bigquery_client.add_dep_to_table(dep_name, curr_ver, get_date(), is_current_using=True)
+    if latest_release_date is None:
+      bigquery_client.add_dep_to_table(dep_name, latest_ver, get_date(), is_current_using=False)
+      latest_release_date = get_date()
+  except Exception:
+    raise
+  return curr_release_date, latest_release_date
+
+
+def compare_dependency_release_dates(curr_release_date, latest_release_date):
+  """
+  Compare release dates of current using version and the latest version.
+  Return true if the current version is behind over 60 days.
+  Args:
+    curr_release_date
+    latest_release_date
+  Return:
+    boolean
+  """
+  if curr_release_date is None or latest_release_date is None:
+    return True
+  else:
+    if (latest_release_date - curr_release_date).days >= _MAX_STALE_DAYS:
+      return True
+  return False
```

Comment:
Consider patch version ?

Reply:
Not here. If we decide to consider the patch version, I will add it into "compare_dependency_versions(...)"